internal/host/hostutil/host_utils.go

```diff
@@ -68,6 +68,19 @@ func ValidateHostname(hostname string) error {
 	return nil
 }
 
+func GetInventoryDisk(host *models.Host, diskId string) (*models.Disk, error) {
+	var inventory models.Inventory
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		return nil, err
+	}
+
+	result := funk.Find(inventory.Disks, func(disk models.Disk) bool {
+		return disk.ID == diskId
```

Comment:
As part of the backwards compatibility concerns I raised above, this function should probably do something like:
If the ID didn't correspond to any disk, it should try comparing it to disk.Path. This way, hosts that had their installation path determined before the new changed rolled out, would still find a disk with this function.

Reply:
Agreed. But having migration script to change the content to installation_disk_path can resolve this issue without need for convoluted code in runtime

Reply:
Or go with the extra new field. Each of the options is better than changing the code here

Reply:
> Agreed. But having migration script to change the content to installation_disk_path can resolve this issue without need for convoluted code in runtime

I'm not sure that's really an option -there are 3 service pods running in production:

`Old`,`Old`,`Old`

In the new version you will migrate (during the service startup I presume) the entire database to use the ID as the installation path rather than path as installation path.

K8s will begin rolling out the new service:

`New`,`Old`,`Old`

While 2 `Old` instances are running, `New` will begin changing the entire database during its startup, while `Old` and `Old are unaware of this change. `Old` and `Old` are still serving requests and they might serve an `install` request for a host already changed by `New` and this might cause problems.

Another issue that might happen is that `Old` and `Old` create new clusters / hosts while `New` is performing the migration - and `New` will miss those and they won't get migrated.

Reply:
Than the best option is to add a new column. It will also solve the compatibility problem of old agents connecting to the new service. Both columns should be maintained in the transition period and than we shall phase out the disk_installation_path

Reply:
please see https://issues.redhat.com/browse/MGMT-4049